types/sanctuary/index.d.ts

```diff
@@ -196,49 +205,40 @@ declare namespace Sanctuary {
     pipe<A, B, C, D, E>(fs: [Fn<A, B>, Fn<B, C>, Fn<C, D>, Fn<D, E>]): (x: A) => E;
     pipe<A, B, C, D, E, F>(fs: [Fn<A, B>, Fn<B, C>, Fn<C, D>, Fn<D, E>, Fn<E, F>]): (x: A) => F;
     pipe(fs: ReadonlyArray<Fn<any, any>>): (x: any) => any;
+    pipeK<B>(fs: ReadonlyArray<Fn<any, Chain<any>>>): <A>(chain_: Chain<A>) => Chain<B>;
     on<A, B, C>(p: Fn2<B, B, C>): (q: Fn<A, B>) => (r: A) => Fn<A, C>;
     //  TODO: Maybe
     isNothing(p: Maybe<any>): boolean;
     isJust(p: Maybe<any>): boolean;
     fromMaybe<A>(p: A): (q: Maybe<A>) => A;
     fromMaybe_<A>(p: Thunk<A>): (q: Maybe<A>) => A;
     maybeToNullable<A>(p: Maybe<A>): Nullable<A>;
-    toMaybe<A>(p: A | null | undefined): Maybe<A>;
     maybe<B>(p: B): <A>(q: Fn<A, B>) => (r: Maybe<A>) => B;
     maybe_<B>(p: Thunk<B>): <A>(q: Fn<A, B>) => (r: Maybe<A>) => B;
     justs<A>(p: ReadonlyArray<Maybe<A>>): A[];
     mapMaybe<A>(p: Fn<A, Maybe<any>>): (q: A[]) => A[];
-    encase<A, B>(p: Fn<A, B>): Fn<A, Maybe<B>>;
-    encase2<A, B, C>(p: Fn2<A, B, C>): Fn2<A, B, Maybe<C>>;
-    encase3<A, B, C, D>(p: Fn3<A, B, C, D>): Fn3<A, B, C, Maybe<D>>;
+    encase<E, A, B>(throwing: Fn2<E, A, B>): (a: A) => Either<E, B>;
```

Comment:
```suggestion
    encase<E, A, B>(throwing: Fn<A, B>): (a: A) => Either<E, B>;
```

Note that `encase` takes a unary function as its first argument, and that `E` appears from nowhere in the return type. `E` is the type of the values the unary function may throw.